scripts/transcript_summarizer_aws_lambda.py

```diff
@@ -1,117 +1,209 @@
+    human_prompt = HumanMessagePromptTemplate.from_template(human_template)
+    chat_prompt = ChatPromptTemplate.from_messages([system_prompt, human_prompt])
+
+    result = chat_model(chat_prompt.format_prompt(transcript=transcript).to_messages())
+    return result.content
+
+
+def qa2essay(transcript, summarized_essay, chat_model=chat):
+    system_template = "You are a helpful assistant that summarizes a transcript of Questions and Answers section of podcasts or lectures."
+    system_prompt = SystemMessagePromptTemplate.from_template(system_template)
+    human_template = (
+        """ You are given a transcript that is a conversation consisting questions from a host and answer from a speaker. \
+  Create a list of question and answeres from it with the following instructions:\
+  Some questions might be spread across multiple lines, so summarise them.\
+  Write a list of questions and answers by respectively tagging Question: and Answer: and do not use numbers.. \
+  The transcript of this presentation is delimited in triple backticks: ```{transcript}``` Use the summary to elaborate on the answers """
+        + summarized_essay
+    )
+    human_prompt = HumanMessagePromptTemplate.from_template(human_template)
+    chat_prompt = ChatPromptTemplate.from_messages([system_prompt, human_prompt])
+    result = chat_model(chat_prompt.format_prompt(transcript=transcript).to_messages())
+    return result.content
+
 
-  result = chat_model(chat_prompt.format_prompt(transcript=transcript).to_messages())
-  return result.content
+def create_essay_parts(transcript_docs, chat_model=chat):
+    essay_response = ""
+    for i, text in enumerate(transcript_docs):
+        essay = transcript2essay(text.page_content, chat_model=chat)
+        essay_response = f"\n\n#Part {i+1}\n".join([essay_response, essay])
 
-def create_essay_parts(transcript_docs):
-  essay_response = ''
-  for i, text in enumerate(transcript_docs):
-      essay = transcript2essay(text.page_content)
-      essay_response = f'\n\n#Part {i+1}\n'.join([essay_response, essay])
+    return essay_response
 
-  return essay_response
+
+def create_qa_parts(transcript_docs, summarized_essay):
+    essay_response = ""
+    for i, text in enumerate(transcript_docs):
+        essay = qa2essay(text.page_content, summarized_essay)
+        essay_response = f"\n\n#Part {i+1}\n".join([essay_response, essay])
+
+    return essay_response
 
 
 def merge_essays(essays, chat_model=chat):
-  system_template = """You are a helpful assistant that summarizes and \
+    system_template = """You are a helpful assistant that summarizes and \
     processes large text information."""
-  system_prompt = SystemMessagePromptTemplate.from_template(system_template)
+    system_prompt = SystemMessagePromptTemplate.from_template(system_template)
 
-  human_template = """Consolidate the multiple parts of the text into one \
+    human_template = """Consolidate the multiple parts of the text into one \
```

Comment:
change to:

`   human_template = """Consolidate the multiple parts of the text into one \
  coherent essay or article that accurately captures the content of the multiple\
  parts without losing any information. The entire text is delimited in triple backticks and the parts are divided by   #heading:\n
  ```{essays}```"""`